src/main/java/com/hms/flexyosisoftconnector/OSIsoftConnectorMain.java

```diff
@@ -125,22 +125,22 @@ private static void getNewPayloadToSend() {
         // Get the data points for current time period in time tracker
         queuePoints = HistoricalDataQueueManager.getFifoNextSpanDataAllGroups(false);
       }
-    } catch (IOException e) {
-      Logger.LOG_EXCEPTION(e);
-      Logger.LOG_WARN(
-          "Exception thrown when obtaining new batch of datapoints. The queue will retry.");
-    } catch (CircularizedFileException e) {
-      Logger.LOG_EXCEPTION(e);
-      Logger.LOG_WARN("Data in the historical log has been overwritten with newer data.");
-    } catch (EbdTimeoutException e) {
-      Logger.LOG_EXCEPTION(e);
-      Logger.LOG_WARN("EBD call has timed out while fetching data.");
-    } catch (JSONException e) {
-      Logger.LOG_EXCEPTION(e);
-      Logger.LOG_WARN("Invalid JSON received while fetching data.");
-    } catch (TimeTrackerUnrecoverableException e) {
+    } catch (Exception e) {
+      String errorMessage = "There was an unknown error collecting data points!";
+      if (e instanceof IOException) {
+        errorMessage =
+            "Exception thrown when obtaining new batch of datapoints. The queue will retry.";
+      } else if (e instanceof CircularizedFileException) {
+        errorMessage = "Data in the historical log has been overwritten with newer data.";
+      } else if (e instanceof EbdTimeoutException) {
+        errorMessage = "EBD call has timed out while fetching data.";
+      } else if (e instanceof JSONException) {
+        errorMessage = "Invalid JSON received while fetching data.";
+      } else if (e instanceof TimeTrackerUnrecoverableException) {
+        errorMessage = "Time tracker file has been corrupted.";
+      }
+      Logger.LOG_WARN(errorMessage);

```

Comment:
This is ok, but it is more standard and easier to follow to see multiple catch statements. In the case of specific exceptions and a generic exception, you would place the `catch (Exception e)` at the end of the list of other `catch` statements

Reply:
I switched to if else statements because it was getting excessively long for what it was. My opinion was that the if else statements were a cleaner solution for more than 1 or 2 catch blocks

Reply:
@oliver-walker-hms  In this senario multiple catch blocks is the conventional/best approach. Try to avoid multiple if,else. A switch would be preferred, but I don't think Java 1.4 can  switch on things other than integers.